src/main/scala/org/broadinstitute/dsde/agora/server/model/AgoraEntityBsonSupport.scala

```diff
@@ -0,0 +1,57 @@
+package org.broadinstitute.dsde.agora.server.model
+
+import com.mongodb.DBObject
+import org.bson.types.ObjectId
+import org.joda.time.DateTime
+import org.joda.time.format.ISODateTimeFormat
+
+
+object AgoraEntityBsonSupport {
+
+  def read(obj: DBObject): AgoraEntity =
+    AgoraEntity(
+      namespace       = stringOrNone(obj, "namespace"),
+      name            = stringOrNone(obj, "name"),
+      snapshotId      = intOrNone(obj, "snapshotId"),
+      snapshotComment = stringOrNone(obj, "snapshotComment"),
+      synopsis        = stringOrNone(obj, "synopsis"),
+      documentation   = stringOrNone(obj, "documentation"),
+      owner           = stringOrNone(obj, "owner"), // Not in DB but checked by tests. Just rolling with this for now.
+      createDate      = dateOrNone(obj, "createDate"),
+      payload         = stringOrNone(obj, "payload"),
+      payloadObject   = None, // Not in DB
+      url             = None, // Not in DB
+      entityType      = entityType(obj),
+      id              = Option(new ObjectId(obj.get("_id").toString)),
+      methodId        = idOrNone(obj, "methodId"), // Configs only
+      method          = None, // Not in DB
+      managers        = Seq.empty, // Not in DB
+      public          = None, // Not in DB
+    )
+
+  private def idOrNone(obj: DBObject, key: String): Option[ObjectId] = {
+    if (obj.containsField(key)) Option(new ObjectId(obj.get(key).toString)) else None
```

Comment:
Is there a reason to convert the retrieved objects to `String` and then back? Are they ever anything but `ObjectId`? This line effectively looks to me like:

```scala
if (obj.containsField(key)) Option(new ObjectId(obj.get(key).asInstanceOf[ObjectId].toString)) else None
```

when it could be:

```scala
if (obj.containsField(key)) Option(obj.get(key).asInstanceOf[ObjectId]) else None
```